frontend/javascripts/oxalis/model/reducers/skeletontracing_reducer_helpers.js

```diff
@@ -72,8 +72,10 @@ function getMaximumNodeId(trees: TreeMap | MutableTreeMap): number {
 }
 
 export function getMaximumTreeId(trees: TreeMap | MutableTreeMap): number {
-  const maxTreeId = _.max(_.map(trees, "treeId"));
-  return maxTreeId != null ? maxTreeId : Constants.MIN_TREE_ID - 1;
+  return Object.values(trees).reduce(
+    (maxId, tree) => (tree.treeId > maxId ? tree.treeId : maxId),
+    Constants.MIN_TREE_ID - 1,
+  );
```

Comment:
Did you also compare this to a simple for-loop? I think, I remember a case where this was another large performance boost. But maybe only because the aggregator was an object in that particular case. For a simple number it might be without benefit. Maybe worth to test, though.

Also: Did you compare line 76 to `Math.max(tree.treeId, maxId)`? If your current way is faster, that's fine, but maybe there is no difference. In that case, I'd prefer the math.max version :)

Reply:
Very good remarks! I did some further testing and also noticed that I worked with a trees array (instead of an object) in my benchmarks :confused: . When using an object there doesn't seem to be a significant speedup over the existing lodash method using reduce or a foor lop (because Object.keys/values needs to be called). I did not spent much more time investigating because not calling `getMaximumTreeId` 20,000+ times provided the much greater speedup and optimizing the `getMaximumTreeId` function seems more like premature optimization at this point :)

Reply:
> When using an object there doesn't seem to be a significant speedup over the existing lodash method using reduce or a foor lop (because Object.keys/values needs to be called).

Did you notice the "not significant speed ups" while micro benchmarking `getMaximumTreeId` or while e2e-benchmarking the entire nml import? Since `getMaximumTreeId` is also used in the rest of WK core, I'd find it worth it to swap the implementation to reduce/for-loop in case there's at least some speed-up factor :)

Reply:
I micro-benchmarked `getMaximumTreeId` and smaller speedups are noticeable only when calling `getMaximumTreeId` >10,000 times for a tracing with at least 20,000 trees. As we don't call it nearly that often (after my fix in this PR), I'm not convinced we need to optimize the function.

Reply:
Cool, totally agree then :+1: